base/linalg/arpack.jl

```diff
@@ -12,20 +12,20 @@ function aupd_wrapper(T, matvecA!::Function, matvecB::Function, solveSI::Functio
                       tol::Real, maxiter::Integer, mode::Integer, v0::Vector)
     lworkl = cmplx ? ncv * (3*ncv + 5) : (sym ? ncv * (ncv + 8) :  ncv * (3*ncv + 6) )
     TR = cmplx ? T.types[1] : T
-    TOL = Array{TR}(1)
-    TOL[1] = tol
+    TOL = Ref{TR}()
+    TOL[] = tol
 
-    v      = Array{T}(n, ncv)
-    workd  = Array{T}(3*n)
-    workl  = Array{T}(lworkl)
-    rwork  = cmplx ? Array{TR}(ncv) : Array{TR}(0)
+    v     = Matrix{T}(n, ncv)
+    workd = Vector{T}(3*n)
+    workl = Vector{T}(lworkl)
+    rwork = cmplx ? Vector{TR}(ncv) : Vector{TR}(0)
 
     if isempty(v0)
-        resid  = Array{T}(n)
```

Comment:
I don't quite understand the logic here with `info` in this function, why does it need to be initialized to zero or one.

Also some of the signatures in arpack look like they can be tightened up to `::Vector`